test/integration/trivial_test.rb

```diff
@@ -23,6 +23,40 @@ def test_copy_data_with_writes_to_source
     assert_test_table_is_identical
   end
 
+  class AlterTableDataWriter
+    include DbHelper
+
+    def initialize
+      @client = Mysql2::Client.new(source_db_config)
+    end
+
+    def start
+      table_name = full_table_name(DEFAULT_DB, DEFAULT_TABLE)
+      @client.query("INSERT INTO #{table_name} VALUES (9000, 'test')")
+      @client.query("ALTER TABLE #{table_name} ADD INDEX (data(100))")
+      @client.query("INSERT INTO #{table_name} (id, data) VALUES (9001, 'test')")
+    end
+  end
+
+  def test_copy_data_with_alter_fails_part_way_through
+    seed_simple_database_with_single_table
+
+    datawriter = AlterTableDataWriter.new
+    ghostferry = new_ghostferry(MINIMAL_GHOSTFERRY)
+
+    ghostferry.on_status(GhostferryHelper::Ghostferry::Status::BINLOG_STREAMING_STARTED) do
+      datawriter.start
+    end
+
+    ghostferry.run_expecting_failure
+
+    source, target = source_and_target_table_metrics
+    source_count = source[DEFAULT_FULL_TABLE_NAME][:row_count]
+    target_count = target[DEFAULT_FULL_TABLE_NAME][:row_count]
+    
+    refute_equal(source_count, target_count, "target should have fewer rows than source")
```

Comment:
`ghostferry.stderr` should also give you the log messages, if you want to match the log message here. I just realized that nothing actually uses it so maybe what you have is fine 🤷